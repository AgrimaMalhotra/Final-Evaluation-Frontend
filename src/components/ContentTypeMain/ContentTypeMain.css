.content-types-main{
  display: flex;
  height: 94vh;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.content-types-column {
  padding: 30px;
  background-color: rgb(206, 206, 222);
}

.column-header {
  width: 15vw;
  display: flex;
  justify-content: space-between;
  color:  #515050;
  margin-bottom: 20px
}

.add-type-button {
  width: 100%;
  height: 5.5%;
  border-radius: 10px;
  border: 2px dotted #7136FA;
  color: #7136FA;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  margin-bottom: 10px;
  font-size: 1.1rem;
}

.content-fields-column {
  background-color:rgb(225, 229, 242);
  width: 100%;
  padding: 30px;
  font-size: 1.8rem;
  font-weight: bolder;
  color: #515050;
}

.content-fields-heading {
  display: flex;
}

.content-fields-heading p {
  margin-right: 10px;
}

.content-fields-heading img {
  width: 20px;
  height: 20px;
  margin-top: 8px;
}

.fields-count {
  font-size: 1.2rem;
  font-weight: 500;
  margin-top: 10px;
  color:  #515050;
  margin-bottom: 20px;
}

.add-field-button {
  width: 100%;
  height: 50px;
  border-radius: 10px;
  border: 2px dotted #7136FA;
  font-family:  Verdana, Geneva, Tahoma, sans-serif;
  color: #7136FA;
  margin-bottom: 20px;
  background: white;
  font-size: 1.1rem;
}