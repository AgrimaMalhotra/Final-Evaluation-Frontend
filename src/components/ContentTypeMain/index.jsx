import React from 'react';
import './ContentTypeMain.css';
import searchIcon from '../../assets/icon-search-dark_2023-03-09/icon-search-dark.png';
import editIcon from '../../assets/user-pencil-write-ui-education_2023-03-09/user-pencil-write-ui-education.png';
import { useState } from 'react';
import NewTypeBar from '../NewTypeBar';
import NewFieldBar from '../NewFieldBar';
import CentreModal from '../CentreModal';

const ContentTypeMain = () => {
  const [hasClickedAddButton, setHasClickedAddButton] = useState(false);
  return (
    <div className="content-types-main">
      <div className='content-types-column'>
        <div className='column-header'>
          <p>7 Types</p>
          <img src={searchIcon} />
        </div>
        <button className='add-type-button' onClick={() => setHasClickedAddButton(true)}>+ New Type</button>
        <div className='type-list' >
          <NewTypeBar />
        </div>
      </div>

      <div className='content-fields-column'>
        <div className='content-fields-heading'>
          <p>Company_Profile</p>
          <img src={editIcon} />
        </div>
        <p className='fields-count'>13 fields</p>
        <button className='add-field-button'>Add another field</button>
        <div className='fields-list'>
          <NewFieldBar />
        </div>
      </div>

      {hasClickedAddButton && <CentreModal setHasClickedAddButton={setHasClickedAddButton} />} 
    </div>
  );
};

export default ContentTypeMain;