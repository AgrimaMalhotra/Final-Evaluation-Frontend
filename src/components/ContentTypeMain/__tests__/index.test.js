import React from 'react';
import {render} from '@testing-library/react';
import ContentTypeMain from '../index';

describe('ContentTypeName', () => {
  const mockContentTypes = [{
    'id': 1,
    'name': 'contentTypeName',
    'fields': {
      'id': 1,
      'name':'abc'}
  }];

  it('should render correctly', () => {
    const component = render(<ContentTypeMain contentTypes={mockContentTypes}/>);
    expect(component).toMatchSnapshot();
  });
});
