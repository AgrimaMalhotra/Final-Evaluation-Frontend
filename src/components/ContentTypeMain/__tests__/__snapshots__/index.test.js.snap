// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentTypeName should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content-types-main"
      >
        <div
          class="content-types-column"
        >
          <div
            class="column-header"
          >
            <p>
              1 Types
               
            </p>
            <img
              src="icon-search-dark.png"
            />
          </div>
          <button
            class="add-type-button"
          >
            + New Type
          </button>
          <div
            class="type-list"
          >
            <div>
              <div
                class="content-type-bar "
              >
                <p>
                  contentTypeName
                </p>
                <p>
                  2
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="content-fields-column"
        >
          <div
            class="content-fields-heading"
          >
            <p />
            <img
              src="user-pencil-write-ui-education.png"
            />
          </div>
          <p
            class="fields-count"
          >
            0 types
          </p>
          <button
            class="add-field-button"
          >
            Add another field
          </button>
          <div
            class="fields-list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content-types-main"
    >
      <div
        class="content-types-column"
      >
        <div
          class="column-header"
        >
          <p>
            1 Types
             
          </p>
          <img
            src="icon-search-dark.png"
          />
        </div>
        <button
          class="add-type-button"
        >
          + New Type
        </button>
        <div
          class="type-list"
        >
          <div>
            <div
              class="content-type-bar "
            >
              <p>
                contentTypeName
              </p>
              <p>
                2
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="content-fields-column"
      >
        <div
          class="content-fields-heading"
        >
          <p />
          <img
            src="user-pencil-write-ui-education.png"
          />
        </div>
        <p
          class="fields-count"
        >
          0 types
        </p>
        <button
          class="add-field-button"
        >
          Add another field
        </button>
        <div
          class="fields-list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
